helm-chart/renku/templates/notebooks/network-policy.yaml

```diff
@@ -1,108 +0,0 @@
-apiVersion: networking.k8s.io/v1
-kind: NetworkPolicy
-metadata:
-  name: {{ template "renku.notebooks.fullname" . }}-k8s-watcher
-spec:
-  podSelector:
-    matchLabels:
-      app: {{ template "renku.notebooks.name" . }}-k8s-watcher
-      release: {{ .Release.Name }}
-  policyTypes:
-    - Ingress
-  ingress:
-    - from:
-        - podSelector:
-            matchLabels:
-              app: {{ template "renku.notebooks.name" . }}
-              release: {{ .Release.Name }}
-        - podSelector:
-            matchLabels:
-              app: renku-data-service
-              release: {{ .Release.Name }}
-      ports:
-        - protocol: TCP
-          port: http
----
-apiVersion: networking.k8s.io/v1
-kind: NetworkPolicy
-metadata:
-  name: {{ template "renku.notebooks.fullname" . }}-ssh-jumphost
-spec:
-  podSelector:
-    matchLabels:
-      app: {{ template "renku.notebooks.name" . }}-ssh
-  policyTypes:
-    - Egress
-  egress:
-    - to:
-        - podSelector:
-            matchLabels:
-              app.kubernetes.io/component: jupyterserver
-              app.kubernetes.io/instance: {{ .Release.Name }}
-              app.kubernetes.io/name: amalthea
-      ports:
-        - port: ssh
-          protocol: TCP
-    - ports:
-        - protocol: UDP
-          port: 53
-        - protocol: TCP
-          port: 53
----
-apiVersion: networking.k8s.io/v1
-kind: NetworkPolicy
-metadata:
-  name: {{ template "renku.notebooks.fullname" . }}-sessions
```

Comment:
I am a bit unsure about the removal of this policy: it is pretty convenient to forbid Sessions to connect around the cluster where they are not supposed to.

Keep in mind that we (and potentially others) manage PG, Redis, Gitlab and other services independently from Renkulab, so the moment we deployed this version of Renkulab we would have to make sure that all those services have corresponding policies to deny user-sessions to connect to them.

Everything else you changed with this PR is great and useful, but I would keep this policy around, for egress only too.

WDYT?

Reply:
So this prevents sessions connecting to anything else in the renku release namespace. But it does not deny access to other namespaces. The problem with the egress policy is the "except internal IPs part". This part is sometimes wrong depending on the cluster setup and it prevents sessions from working because when they try to connect to things outside of the cluster the ips map to the ip ranges that are usually just private/internal.

Reply:
So Alessandro as we discussed over slack. I have added the following:
- deny all egress from user sesssions
- allow all egress form user sessions (but only for requests that target the release namespace) and this is fine because inside the release namespace we deny all ingress by default

Reply:
Okay, sounds good.
Thank you for getting to the bottom of this!